qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -137,11 +151,18 @@ def __init__(
             TypeError: When one of converters has an invalid type.
             QiskitOptimizationError: When the minimum eigensolver does not return an eigenstate.
         """
-
+        if isinstance(min_eigen_solver, LegacyMinimumEigensolver):
```

Comment:
I am not sure whether we need a warning here. The legacy algos in Terra are pending deprecated and will be deprecated/removed, when creating one of these a user should already see a warning about them going away. We can have a warning here too, its just that the user should have already seen they should be using the newer ones.

Reply:
Thank you for your suggestion. I will remove the warning message.